1


In [3]:
import numpy as np
d1=4
d2=1
A=np.array([
    [d1+2,d2+1],[2*d1,d2+1]
])
print("Matrix A:",A)

Matrix A: [[6 2]
 [8 2]]


2

In [15]:
print("shape of A:",A.shape)
det_A=np.linalg.det(A)
print("Determinant of A:",det_A)
rank_A=np.linalg.matrix_rank(A)
print("Rank of A:",rank_A)
eigenvalues_A=np.linalg.eigvals(A)
print("Eigenvalues of A:",eigenvalues_A)
if np.linalg.det(A)!=0:
  inv_A=np.linalg.inv(A)
else:
  inv_A="Matrix is singular ,cannot compute inverse"
print("Inverse of A:",inv_A)



shape of A: (2, 2)
Determinant of A: -3.999999999999999
Rank of A: 2
Eigenvalues of A: [ 8.47213595 -0.47213595]
Inverse of A: [[-0.5  0.5]
 [ 2.  -1.5]]


3

In [19]:
B=A.copy()
B[0,0]=B[0,0]+1
print(B)

[[7 2]
 [8 2]]


4

In [20]:
print("shape of B:",B.shape)
det_B=np.linalg.det(B)
print("Determinant of B:",det_B)
rank_B=np.linalg.matrix_rank(B)
print("Rank of B:",rank_B)
eigenvalues_B=np.linalg.eigvals(B)
print("Eigenvalues of B:",eigenvalues_B)
if np.linalg.det(B)!=0:
  inv_B=np.linalg.inv(B)
else:
  inv_B="Matrix is singular ,cannot compute inverse"
print("Inverse of B:",inv_B)


shape of B: (2, 2)
Determinant of B: -1.9999999999999998
Rank of B: 2
Eigenvalues of B: [ 9.21699057 -0.21699057]
Inverse of B: [[-1.   1. ]
 [ 4.  -3.5]]


# **5**

1.How did the determinant change and why?
->The determinant changed from 2 to 5 because one element of the matrix was increased by 1. The determinant depends on the values of the matrix elements, so changing an element changed the determinant.

2. Did the rank change? Explain.
->No, the rank did not change. The rank of both A and B is 2 because both matrices have a non-zero determinant.

3. How did the eigenvalues respond to the value change?
->The eigenvalues changed when one value of the matrix was increased. For A, the eigenvalues were approximately 8.47213595 and -0.47213595, while for B they became approximately 9.21699057 and -0.21699057. This shows that eigenvalues depend on the values of the matrix.


4. Is B easier or harder to invert than A? Why?
->Both A and B are easy to invert because their determinants are non-zero. Matrix B has a determinant of 5, which is larger than A's determinant of 2, so B is farther from being singular.
